libs/pika/async_cuda/include/pika/async_cuda/cuda_pool.hpp

```diff
@@ -135,25 +156,64 @@ namespace pika::cuda::experimental {
         std::shared_ptr<pool_data> data;
 
     public:
+        /// \brief Construct a pool of CUDA streams and handles.
+        ///
+        /// \param device the CUDA device used for scheduling work
+        /// \param num_normal_priority_streams_per_thread the number of normal priority streams per
+        /// thread
+        /// \param num_high_priority_streams_per_thread the number of high priority streams per
+        /// thread
+        /// \param flags flags used to construct CUDA streams
+        /// \param num_cublas_handles the number of cuBLAS handles to create for the whole pool
+        /// \param num_cusolver_handles the number of cuSOLVER handles to create for the whole pool
         PIKA_EXPORT explicit cuda_pool(int device = 0,
             std::size_t num_normal_priority_streams_per_thread = 3,
             std::size_t num_high_priority_streams_per_thread = 3, unsigned int flags = 0,
             std::size_t num_cublas_handles = 16, std::size_t num_cusolver_handles = 16);
+
+        /// \brief Move constructor.
         PIKA_NVCC_PRAGMA_HD_WARNING_DISABLE
         cuda_pool(cuda_pool&&) = default;
+        /// \brief Copy constructor.
         PIKA_NVCC_PRAGMA_HD_WARNING_DISABLE
         cuda_pool(cuda_pool const&) = default;
+        /// \brief Move assignment operator.
         PIKA_NVCC_PRAGMA_HD_WARNING_DISABLE
         cuda_pool& operator=(cuda_pool&&) = default;
+        /// \brief Copy assignment operator.
```

Comment:
I'm not sure there's any value in documenting these like above. Should we just leave the docstrings out and say it's copyable/movable etc. and describe the semantics in the class docstring?

Reply:
Remove these, unless there's special behaviour to document.